### ISSUE TITLE
Enter Key Not Working Activate Pitch Bend Range Typein

### ISSUE THREAD

**Issue by ironcross32** [surge-synthesizer/surge#7557]
**Bug Description:**
Pressing enter/return on the keyboard does not activate the typein on either the up or down fields of the pitch bend range controls.
**Surge XT Version**
This information is found on the About screen, which you get to from the bottom right menu.
1.3.nightly.d64fdde
* Version:
* Plugin Type: [VST2, VST3, AU]
* Bitness: [32-bit or 64-bit]
VST3 64-bit
**Reproduction Steps:**
1. Go to any of the pitch bend controls
2. Try to press enter or/return to activate the typein
3. Note that it only produces the default OS beep (if any) but does not activate the typein

**Expected Behavior:**
Pressing enter should activate the typein.
**Screenshots:**
If applicable, add screenshots/GIF/videos to help explain your problem.

**Computer Information (please complete the following!):**
 - OS: [e.g. Windows 10]
 - Host: [e.g. Bitwig, Logic, Cubase, Reaper]
 - Version: [e.g. 2.4]
Windows 11; Reaper 7.06
**Additional Information:**

**Comment by mkruselj** [comment]
@baconpaul If you can spare a couple of minutes, what would be the solution here? I see how this is done in ModulatableSlider in keyPressed(), but the same is not working for NumberField:

```
    if (action == Return)
    {
        auto sge = firstListenerOfType<SurgeGUIEditor>();
        if (sge && sge->promptForUserValueEntry(this))
            return true;
    }
```

The problem is that `this` pointer doesn't satisfy `promptForUserValueEntry()` because NumberField is not a subclass of ModulatableControlInterface class. And so then I need a parameter and a component argument. But then not even all NumberFields we use are even declared as Parameters (most notably the two in MSEG Editor).

Wat do?

**Comment by baconpaul** [comment]
I’ll look tomorrow morning. But the trick is going to be to look at the tag and see if it is in the param tag range probably

**Comment by baconpaul** [comment]
OK took a quick look and this could be mildly tricky. I think the user value entries have ever been used on anything other than floats.

If I was gonna tackle it i would do this

1. Add `SurgeGUIEdtiro::promptForUserValueEntry(uint32_t, juce::Component *)` to SGE.h

2. In SGE.cpp implement it as
```
bool SurgeGUIEditor::promptForUserValueEntry(uint32_t tag, juce::Component *comp)
{
    auto t = tag - start_paramtags;
    if (t < 0 || t >= n_total_params)
        return false;

    auto p = synth->storage.getPatch().param_ptr[t];

    return promptForUserValueEntry(p, mci->asJuceComponent());
}
```

From your thing call

```
if (sge && sge->promptForUserValueEntry(getTag(), this)) return true;
```

and that will at least call PFUVE on a parameter but it may still not work if p is type `vt_int`. In that case you just need to debug whichever path is not working.

**Comment by mkruselj** [comment]
Thanks! I'll hack around that. 🙂

**Comment by mkruselj** [comment]
Nice, this works for all except value edits in MSEG editor!

**Comment by baconpaul** [comment]
Yeah the mseg has different problems with ally so if this works then let’s go for it! As long as it doesn’t crash in mseg which the ifs mean it should not

**Comment by mkruselj** [comment]
It doesn't crash but it sucks that it cannot just work as the rest... Do I absolutely have to register it as an accessibility handler, or can I just add a lambda to `h/vSnapSize->keyPressed()` and respond to Return directly there?

**Comment by baconpaul** [comment]
You need a different function altogether to respond to return and giving it an ally handler won’t work anyway since the parent is not accessible by design 

I’ll think about it. We are so closely tied to params in the type in it may be annoying - need to read the code again

let’s merge the Param case though at least. It’s just upside

**Comment by mkruselj** [comment]
OK on it!
